orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/entity/Session.java

```diff
@@ -53,6 +53,22 @@ public Session(String sessionId) {
         initializeCodeRequestMap();
     }
 
+    public Session(Session session, String newSessionId) {
```

Comment:
Not totally convinced on whether this is needed. Seems like it will be only used for one specific case. Could all these fields just be set in `copySessionForMaxAge` instead?

Reply:
Hmm would you prefer a _plain_ copy constructor with the fields being modified in the method instead?

Reply:
Ah yeah I was um-ing and ah-ing about this bit - happy to change!